source/gui/guiHelper.py

```diff
@@ -285,7 +285,7 @@ def addLabeledControl(self, labelText, wxCtrlClass, **kwargs):
 		"""
 		labeledControl = LabeledControlHelper(self._parent, labelText, wxCtrlClass, **kwargs)
 		if(isinstance(labeledControl.control, (wx.ListCtrl,wx.ListBox,wx.TreeCtrl))):
-			self.addItem(labeledControl.sizer, flag=wx.EXPAND)
+			self.addItem(labeledControl.sizer, flag=wx.EXPAND, proportion=1)

```

Comment:
I didn't really think about this with the other PR's; have you thought about how this will affect other controls using `guihelper`? This could affect the layout of other dialogs.

Reply:
> I didn't really think about this with the other PR's; have you thought about how this will affect other controls using `guihelper`? This could affect the layout of other dialogs.

Indeed, it will affect their layout, but for the better.
For one thing, it is mandatory to allow newly resizable dialogs to grow properly.
Setting the `wx.EXPAND` flag with the default proportion of 0 makes the item added to a `wx.BoxSizer` expand only in the same dimension as the sizer, while we most likely want both dimensions for a data control such as those impacted here.
Reference: https://wxpython.org/Phoenix/docs/html/sizers_overview.html#sizers-overview

Furthermore, I've checked all the settings panels, and none is impacted as the panels themselves cannot grow horizontally.
Allowing them to could have some benefit, though, but it is another matter.

Reply:
If all controls are added with proportion=1 then when the dialog grows, the controls will grow at the same rate. We don't have that yet, as not all controls are added with proportion=1, some won't grow. All controls growing may also not be what we want either, sometimes we want one control fixed. It is difficult to get this right for all situations.  

 The hard part with setting expand or proportion is that it indicates this should be "the default for NVDA". The GuiHelper was initially developed to take all of the most common cases and abstract the details. It has a major flaw however, in that it is not very flexible. I have another approach which I hope to introduce soon which should improve this. I think we can go ahead with this PR, if we discover unwelcome side effects later we can special case those.

Reply:
> It is difficult to get this right for all situations.

Indeed.

> The hard part with setting expand or proportion is that it indicates this should be "the default for NVDA". The GuiHelper was initially developed to take all of the most common cases and abstract the details.

This is the point. expandable with proportion 1 does not suit every single complicated case, but IMHO is a pretty reasonable default, especially when one doesn't see the result of the GUI he/she creates.

> It has a major flaw however, in that it is not very flexible. I have another approach which I hope to introduce soon which should improve this.

Looking forward to seeing it.
When mentoring a developer from a customer organization who was implementing GUI part of some add-on, I first instructed "Use the GUI Helper as much as possible." but we soon realized it did not cover enough basic scenarios.

> I think we can go ahead with this PR, if we discover unwelcome side effects later we can special case those.

Sure.